src/sys/socket/addr.rs

```diff
@@ -45,233 +45,434 @@ pub(crate) const fn ipv6addr_to_libc(addr: &net::Ipv6Addr) -> libc::in6_addr {
     }
 }
 
-/// These constants specify the protocol family to be used
-/// in [`socket`](fn.socket.html) and [`socketpair`](fn.socketpair.html)
-///
-/// # References
-///
-/// [address_families(7)](https://man7.org/linux/man-pages/man7/address_families.7.html)
-// Should this be u8?
-#[repr(i32)]
-#[non_exhaustive]
-#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
-pub enum AddressFamily {
-    /// Local communication (see [`unix(7)`](https://man7.org/linux/man-pages/man7/unix.7.html))
-    Unix = libc::AF_UNIX,
-    /// IPv4 Internet protocols (see [`ip(7)`](https://man7.org/linux/man-pages/man7/ip.7.html))
-    Inet = libc::AF_INET,
-    /// IPv6 Internet protocols (see [`ipv6(7)`](https://man7.org/linux/man-pages/man7/ipv6.7.html))
-    Inet6 = libc::AF_INET6,
-    /// Kernel user interface device (see [`netlink(7)`](https://man7.org/linux/man-pages/man7/netlink.7.html))
-    #[cfg(linux_android)]
-    Netlink = libc::AF_NETLINK,
-    /// Kernel interface for interacting with the routing table
-    #[cfg(not(any(linux_android, target_os = "redox")))]
-    Route = libc::PF_ROUTE,
-    /// Low level packet interface (see [`packet(7)`](https://man7.org/linux/man-pages/man7/packet.7.html))
+/// A possible error when converting a c_int to [`AddressFamily`].
+#[derive(Debug, Clone, Copy)]
+pub struct InvalidAddressFamilyError;
+
+/// Address families, corresponding to `AF_*` constants in libc.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct AddressFamily(libc::c_int);
```

Comment:
All Newtypes should use repr(transparent).  You might even be able to cast directrly from AddressFamily to other int types just by using `as`, instead of `.family()`.
```suggestion
#[repr(transparent)]
pub struct AddressFamily(libc::c_int);
```